### PR TITLE
Add support for continuous action spaces to PCN

### PR DESCRIPTION
**Pull request by vaidas-sl**
[Original paper](https://arxiv.org/pdf/2204.05036.pdf) suggests that PCN could be used in continuous action space. [This](https://arxiv.org/pdf/2204.05027.pdf) paper uses PCN in continuous action space. This PR changes PCN implementation so it would be usable in cont actions space. Also updated constructor to accept model_class, to make it easier to pass different models, as original paper uses different models for different tasks

### CODE REVIEW

**Review by ffelten — COMMENTED** [review]
I made a quick pass, mostly looks good to me. I guess you should also update the algorithms table in the README :-). 

Also, @LucasAlegre or @mathieu-reymond will probably be more knowledgeable than me on these changes.

File: morl_baselines/common/evaluation.py [hunk 1]
```diff
@@ -178,10 +180,23 @@ def log_all_multi_policy_metrics(
...
     )
-    front = wandb.Table(
-        columns=[f"objective_{i}" for i in range(1, reward_dim + 1)],
-        data=[p.tolist() for p in filtered_front],
-    )
+    columns = [f"objective_{i}" for i in range(1, reward_dim + 1)]
+    data = [p.tolist() for p in filtered_front]
+
+    # Filter the horizons array using filtered_front so that filtered horizons would contain horizons for the filtered front
+    if horizons is not None:
```
[the remark this answers is not in the document]
Comment (LucasAlegre) [reply morl_baselines/common/evaluation.py:187]: Removed

File: morl_baselines/multi_policy/pcn/pcn.py [hunk 2]
```diff
@@ -135,8 +162,18 @@ def __init__(
         self.scaling_factor = scaling_factor
         self.desired_return = None
         self.desired_horizon = None
+        self.continuous = continuous
```
[the remark this answers is not in the document]
Comment (LucasAlegre) [reply morl_baselines/multi_policy/pcn/pcn.py]: Done

**Comment by LucasAlegre** [comment]
@vaidas-sl I fixed the noise added to the action at training time, as defined in the paper. I also removed the change to the log_all_multi_policy_metrics, as it is supposed to be a method for evaluating Pareto fronts, and I believe that the metric you want should be logged inside PCN's code only.

**Review by LucasAlegre — APPROVED** [review]

**Pull request merged**